drivers/driver-test/src/commonMain/kotlin/com/squareup/sqldelight/driver/test/DriverTest.kt

```diff
@@ -176,3 +170,13 @@ abstract class DriverTest {
     }
   }
 }
+
+private fun SqlDriver.changes(): Long? =
+        // wrap in a transaction to ensure read happens on transaction thread/connection
+        object : TransacterImpl(this) {}.transactionWithResult {
```

Comment:
`newTransaction()` ?

Reply:
Ah! I was used to not having access to the internal APIs needed to end the transaction, but I can do that here.

Reply:
actually, i can't call endTransaction for some reason. once i start a transaction, how do i end it without a Transacter?

Reply:
```
Cannot access 'endTransaction': it is internal in 'Transaction
```

Reply:
I think it's because `Transaction` stuff is in `runtime`, and this test in in native-driver.

Reply:
try casting it as a `NativeSqlDriver.Transaction`

Reply:
ah that class it private

Reply:
why does this need to be in a transaction at all? it confused me

Reply:
changes operates per connection https://sqlite.org/c3ref/changes.html

Reply:
did you try without the transaction and it wasn't working? my assumption was that changes would still be fine across connections/threads

Reply:
Yep it fails now without thus update.

Reply:
Presumably then the reverse is true. If I remove the driver change, this test will fail again.

Reply:
got it. Can we take `object : TransacterImpl(this)` and put it in a constant private val for this test then?

Reply:
Yup! Will do this morning.

Reply:
Done. Also confirmed that the reverse is true.